src/services/api/addresses.ts

```diff
@@ -154,7 +154,7 @@ export async function getToken(
   token: string,
   owner: Address
 ) {
-  const jettonWalletKey = `${token}:jettonWallet`;
+  const jettonWalletKey = `${token}:jettonWallet:${owner}`;
```

Comment:
This is enough to solve the bug.

The reason I still think it's a good idea to throw (or better, refactor API such that `getOwner()` can return null) is that it would still be confusing to inspect the cache and see an entry for the zero address there